src/components/chat/chat-room-search-input.tsx

```
@@ -0,0 +1,35 @@
+import { useEffect, useState } from "react";
+
+export default function ChatRoomSearchInput() {
+  const tabType = useChatRoomStore((s) => s.tabType);
+  const searchQuery = useChatRoomStore((s) => s.searchQuery);
+  const setSearchQuery = useChatRoomStore((s) => s.setSearchQuery);
+
+  const [localValue, setLocalValue] = useState(searchQuery);
+
+  // tab 변경 시 store의 searchQuery가 ""로 초기화됨 → 로컬 input도 동기화
+  useEffect(() => {
+    setLocalValue(searchQuery);
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

/home/jailuser/git/src/components/chat/chat-room-search-input.tsx:17:5
  15 |   // tab 변경 시 store의 searchQuery가 ""로 초기화됨 → 로컬 input도 동기화
  16 |   useEffect(() => {
> 17 |     setLocalValue(searchQuery);
     |     ^^^^^^^^^^^^^ Avoid calling setState() directly within an effect
  18 |   }, [tabType, searchQuery]);
  19 |
  20 |   const handleSubmit = () => {

(react-hooks/set-state-in-effect)